src/types.ts

```diff
@@ -24,19 +11,32 @@ export interface Tags {
   };
 }
 
+export interface InvestmentOpportunity {
+  readonly divest: DivestScript;
+  readonly dataURI: string; // API endpoint to query for user data from, e.g. balances
+  readonly invest: InvestScript;
+  readonly logoURI?: string;
+  readonly strategy?: StrategyLeg[];
+  readonly tokens: {
+    // arrays of token addresses, to be looked up in the `tokens` mapping
+    readonly input: string[]; // token addresses the user will deposit to enter this opportunity, and receipt upon exiting
+    readonly receipt: string[]; // receipt token addresses the user needs to exit this opportunity
+    readonly rewards: string[]; // rewards token addresses the user may claim when exiting this opportunity
+  };
+}
+
 // Parameters for investing into a protected investment.
 //
-// The function signature is constant, which is why it's an optional property. The `tokens` property defines the
-// set of input tokens that can be used to deposit into this investment opportunity. The items in the function
-// signature correspond to these `tokens`. For example, if we're investing in the 3Crypto pool, the `tokens` array
-// would be [ETH, WBTC, USDT]. When calling `invest` on `address`, amounts[0] would correspond to the amount of ETH
-// to invest, amounts[1] would correspond to the amount of WBTC to invest, etc. An amount of MAX_UINT256 should be
-// used to signal that you want to invest all tokens you have
+// Example  'function invest(uint256[] calldata amounts) external payable'
```

Comment:
More realistic example based on current invest signatures
```suggestion
// Example: 'function invest(address market, uint256 amount) external payable'
```

Reply:
much better, thx